addons/mail/static/src/discuss/core/common/thread_model_patch.js

```diff
@@ -322,6 +322,17 @@ const threadPatch = {
         if (this.isChatChannel && this.selfMember?.message_unread_counter_ui) {
             return this.selfMember.message_unread_counter_ui;
         }
+        if (this.discussAppCategory?.id === "channels") {
+            if (this.store.settings.channel_notifications === "no_notif") {
+                return 0;
+            }
+            if (
+                this.store.settings.channel_notifications === "all" &&
+                !this.selfMember?.mute_until_dt
+            ) {
+                return this.selfMember?.message_unread_counter_ui;
+            }
+        }
         return super.importantCounter;
     },
```

Comment:
Nitpick:

Instead of dedicated branching for channels, I would make it use the same branching code as with chat.
Functionally the channels with "all notifs" behave like chat notifications, so this would make more sense to share the same code.

The code above we could replace the `this.isChatChannel` part to another function that can be slightly customized to take `channel_notifications` into account.

Reply:
The new function would have semantics of `areAllUnreadMessagesImportant`, for which "chat" conversations is intrinsically and for channels it's when settings say so.

Using the same branching makes it less likely to be big, as a test for chat that works is a big part for channels to work too. With a dedicate branching, a test becomes a necessity.

I'm not against tests, but at the same time it's unreasonable to expect to have a test for everything, and tests have a maintenant cost. At the end of the day we want to write code fast and minimise risks of bugs, so would be nice to have chat test to contributes significantly to correctness of the channel code.

Not a huge deal, really the code is good enough as it is with this diff.